### DISCUSSION TITLE
allow to set a value once – but disable editing

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#5683] [Feature Requests]
Oftentimes we need to be able to (manually) set a value but we need to make sure that value is not edited/changed later. 

We can already set a field to "readonly". The option in the app says: "disable editing value".

I propose to split this into two options:
- readonly (can never be changed or manually set)
- disable editing value (can be set once and not altered later)

Or rather:
- allow setting/adding
- allow editing

This would allow for greater flexibility and should be useful in a lot of use case.

**Comment by rijkvanzanten** [comment]
I'm not sure if this is a field-configuration thing or a permissions thing 🤔 Disabling the field in the update permissions already achieves the same goal 🙂

**Reply by ghost** [reply]
I'm aware of that approach but I think when working with data there's a difference between a role-based and a data-model-based perspective. What I'm saying is not that certain user roles should not be able to edit values, but that those values cannot be changed from a logical perspective. 

I can see though that user roles are very powerful. It's also the only place in the app where I can find field validation rules, right? So, yes that's how I'd implement this for now.

I still think it'd be good to move (copy) some of that logic to the items/fields themselves, in order to make it easier to understand and maintain a data model. Technically, what you're saying means that we could remove the "readonly" and "hidden" options from fields and just set permissions accordingly. Yes, maybe … but it just feels like the 'wrong' (or: not the only) place for that kind of stuff :)

**Comment by superhp** [comment]
I want to raise this idea again. It would make a lot of sence to make read only settable once (as it is for readonly properties in Typescript as well). With current readonly behavior, there might be some issues. For example, you create a new collection with a required readonly field. And just after you created it, you already are not allowed to create items in that collection because you can't set required readonly field. It is not a good option to remove readonly every time you want to create a new item and content managers shouldn't even have access to collection configuration. Role permission settings can't help in this situation as well. Do you think, the readonly behavior could be changed so that it can be set once?

**Reply by azrikahar** [reply]
This might not be a direct answer, but one alternative I can think of to make `readonly` fields can be set on create is to actually do:

- have a field that will be auto-populated on create, such as `date_created`
- opt to set a condition where if `date_created` is NOT filled, which is during create only, then it should remove `readonly` from input field

Here's a short clip on the above description:

https://user-images.githubusercontent.com/42867097/148522478-a6cfde2f-617b-47e5-aedc-eec9136c9b62.mp4

Hope that helps to achieve your current use case here.

**Reply by superhp** [reply]
I think, this solution could work! :)

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
